### ISSUE TITLE
MEV-Boost Direct builder connection

### ISSUE THREAD

**Issue by quintuskilbourn** [flashbots/mev-boost#648]
Currently, relays are in a critical path as bids flow from builders to proposers. This adds a little latency and places relays in a (permissioned) position to significantly impact the functioning of the (permissionless) block auction. However, looking at optimistic relaying, we see that relays often don't need to be in this privileged position.

I wrote a small proposal for how MEV-Boost could be changed to skip relays where they are not necessary by first querying information about optimistic builder endpoints and then interacting with those directly. This change does not require CL or EL changes.

More details [here](https://collective.flashbots.net/t/mb-direct-an-idea-for-mev-boost-improvement/3350?u=quintus)

![image](https://github.com/flashbots/mev-boost/assets/47097067/58d0ba88-eb49-458f-8ee2-0338d2a164ca)

**Comment by avalonche** [comment]
in this model, wouldn't the validator's IP information be exposed from the incoming getHeader request to the builder?

**Comment by ralexstokes** [comment]
> in this model, wouldn't the validator's IP information be exposed from the incoming getHeader request to the builder?

yes, unless they take some steps to obscure this -- the relay today acts as a bit of a shield privacy-wise as proposers already trust relays to some extent and so we can justify this additional marginal ask (acting as privacy provider)

**Comment by quintuskilbourn** [comment]
Yeah this is the most shaky part. You can probably back out the proposers IP after it has proposed a few blocks. 

If we could get some kind of IP rotation or similar it wouldn’t be a problem. Open to ideas 

another fun idea is to have some other entities (maybe relays) create noise that correlates with validators to obscure their IP.

**Comment by avalonche** [comment]
as the endpoint is unauthenticated, builders will probably obscure the validator IP when querying for each other's bids. Its possible that builders could ddos the getHeader endpoint to prevent proposers from getting a competitor bid and end up having to submit through the relay.

**Comment by quintuskilbourn** [comment]
I don’t see why the DoS concern is any different than with relay endpoints?

**Comment by Champ10q** [comment]
I hope i got this
